website_sale_loyalty_suggestion_wizard/templates/promotion_templates.xml

```diff
@@ -0,0 +1,185 @@
+                                    class="btn-close"
+                                    data-bs-dismiss="modal"
+                                    aria-label="Close"
+                                />
+                            </div>
+                            <div class="modal-body" id="o_promo_config_body">
+                                <div
+                                    t-if="request.session.error_promo_code"
+                                    class="alert alert-danger"
+                                    role="alert"
+                                >
+                                    <span t-out="request.session.error_promo_code" />
+                                </div>
+                                <t
+                                    t-call="website_sale_loyalty_suggestion_wizard.configure"
+                                />
+                                <span
+                                    class="d-none js_promotion_change"
+                                    t-att-data-promotion-id="promotion_id and promotion_id.id or ''"
+                                />
+                                <span
+                                    class="d-none website_sale_order_id"
+                                    t-att-data-order-id="website_sale_order and website_sale_order.id or ''"
+                                />
+                            </div>
+                            <div class="modal-footer">
+                                <button
+                                    type="button"
+                                    class="btn btn-primary o_coupon_selection_wizard_apply"
+                                >Apply</button>
+                                <button
+                                    type="button"
+                                    class="btn btn-light"
+                                    data-bs-dismiss="modal"
+                                >Close</button>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+            </t>
+        </xpath>
+    </template>
+    <!-- List of suggested promotions in shopping cart -->
+    <template
+        id="website_sale_loyalty_suggestion"
+        inherit_id="website_sale.cart_lines"
+        active="True"
+        customize_show="True"
+        name="Promotions suggestions"
+    >
+        <xpath expr="//th[hasclass('td-img')]" position="before">
+            <t
+                t-set="promotions"
+                t-value="website_sale_order.website_order_line.suggested_promotion_ids.filtered(lambda x: x.is_published)[:1].id"
+            />
+            <t t-if="promotions">
+                <th />
+            </t>
+        </xpath>
+        <xpath expr="//td[hasclass('td-img')]" position="before">
+            <td class="td-promo-suggestion" t-if="promotions">
+                <t t-set="promo" t-value="line.suggested_promotion_ids[:1].id" />
```

Comment:
**I tested this case:**

I created two rules: one for the Storage Box and another for the Large Cabinet.
Both products are displayed on the website.

- Add both products to the cart.
- Go to the cart, and the first promotion sale-promotion (related to the Storage Box) is suggested. Close the wizard.
- In the cart, on the line with the Large Cabinet product, click the button to apply the promotion.
- The wizard displays the promotion related to the Storage Box product, not the Large Cabinet.

I expected the wizard to display the promotion related to the selected product.
I’ve attached a video for more details. Is this behavior expected?
![sale-promotion](https://github.com/user-attachments/assets/398892ed-12ad-4803-bb07-109548ea34f9)

Reply:
It should be resolved by now